vertx-template-engines/vertx-web-templ-handlebars/src/main/java/io/vertx/ext/web/templ/impl/HandlebarsTemplateEngineImpl.java

```diff
@@ -70,6 +70,13 @@ public void render(RoutingContext context, String templateFileName, Handler<Asyn
       Template template = cache.get(templateFileName);
       if (template == null) {
         synchronized (this) {
+          // Prepare templateDirectory for partials
+          if (loader.getTemplateDirectory() == null) {
+            loader.setTemplateDirectory(templateFileName.substring(0,
+              // preserve '/' as Utils#normalizePath guarantees a leading slash
+              templateFileName.length() - Utils.pathOffset(context.normalisedPath(), context).length() + 1
```

Comment:
Ideally templateDirectory would be passed directly from TemplateHandlerImpl, but that requires an API addition to TemplateEngine or special case handling for handlebars in TemplateHandlerImpl. The substring works, though.